crates/subspace-farmer/src/archiving.rs

```diff
@@ -94,27 +97,50 @@ impl Archiving {
             ControlFlow::Continue(())
         };
 
-        // Erasure coding in archiver and piece encoding are CPU-intensive operations.
-        tokio::task::spawn_blocking({
-            move || {
-                let mut last_archived_segment_index = None;
-                while let Ok((archived_segment, acknowledgement_sender)) =
-                    archived_segments_sync_receiver.recv()
-                {
-                    let segment_index = archived_segment.root_block.segment_index();
-                    if last_archived_segment_index == Some(segment_index) {
-                        continue;
+        if let Some(node) = node {
+            tracing::trace!("Subscribing to pubsub archiving...");
+            let mut subscription = node
+                .subscribe(subspace_networking::PUB_SUB_ARCHIVING_TOPIC.clone())
+                .await?;
```

Comment:
As soon as network produces more pieces than farmer  can (for whatever reason) process, farmer will run out of memory and crash. There should be graceful handling for overload or at least a loud TODO about this. Without verification especially this is a very good DoS vector for farmer and such sync should not be used in production. Corresponding flag should be clearly marked as experimental and not recommended for use.